packages/q/qt5base/xmake.lua

```diff
@@ -0,0 +1,216 @@
+                if package:is_targetarch("x86_64", "x64") then
+                    arch = "android_x86_64"
+                elseif package:is_targetarch("arm64", "arm64-v8a") then
+                    arch = "android_arm64_v8a"
+                elseif package:is_targetarch("armv7", "armv7-a") then
+                    arch = "android_armv7"
+                elseif package:is_targetarch("x86") then
+                    arch = "android_x86"
+                end
+            else
+                arch = "android"
+            end
+        end
+
+        local installdir = package:installdir()
+        os.vrunv("aqt", {"install-qt", "-O", installdir, host, target, version:shortstr(), arch})
+
+        -- move files to root
+        local subdirs = {}
+        if package:is_plat("linux") then
+            table.insert(subdirs, package:is_targetarch("x86_64") and "gcc_64" or "gcc_32")
+            table.insert(subdirs, package:is_targetarch("x86_64") and "clang_64" or "clang_32")
+        elseif package:is_plat("macosx") then
+            table.insert(subdirs, package:is_targetarch("x86_64") and "clang_64" or "clang_32")
+        elseif package:is_plat("windows") then
+            import("core.project.config")
+            local vs = config.get("vs")
+            if vs then
+                table.insert(subdirs, package:is_targetarch("x64") and "msvc" .. vs .. "_64" or "msvc" .. vs .. "_32")
+                table.insert(subdirs, "msvc" .. vs)
+            end
+            table.insert(subdirs, package:is_targetarch("x64") and "msvc*_64" or "msvc*_32")
+            table.insert(subdirs, "msvc*")
+        elseif package:is_plat("mingw") then
+            table.insert(subdirs, package:is_targetarch("x86_64") and "mingw*_64" or "mingw*_32")
+        elseif package:is_plat("android") then
+            local subdir
+            if package:is_targetarch("arm64-v8a") then
+                subdir = "android_arm64_v8a"
+            elseif package:is_targetarch("armeabi-v7a", "armeabi", "armv7-a", "armv5te") then -- armv7-a/armv5te are deprecated
+                subdir = "android_armv7"
+            elseif package:is_targetarch("x86", "i386") then -- i386 is deprecated
+                subdir = "android_x86"
+            elseif package:is_targetarch("x86_64") then
+                subdir = "android_x86_64"
+            end
+            if subdir then
+                table.insert(subdirs, subdir)
+            end
+            table.insert(subdirs, "android")
+        elseif package:is_plat("wasm") then
+            table.insert(subdirs, "wasm_32")
+        else
+            table.insert(subdirs, "*")
+        end
+
+        local installeddir
+        for _, subdir in pairs(subdirs) do
+            local results = os.dirs(path.join(installdir, version, subdir), function (file, isdir) return false end)
+            if results and #results > 0 then
+                installeddir = results[1]
+                break
+            end
+        end
```

Comment:
I think that we should use `lib.detect.find_directory` to find install directory, it will be better.

Reply:
Or we can use find_file to find qmake path and get it's install root directory. like find_qt.lua

https://github.com/xmake-io/xmake/blob/bcf29f16d1ffd65b07a11e80e08ebac42eef4083/xmake/modules/detect/sdks/find_qt.lua#L118-L121